### DISCUSSION TITLE
What is the difference between router vs expo-router?

### DISCUSSION THREAD

**Discussion by jsb188** [expo/router#784] [Q&A]
expo-router package exports "router" and "useRouter". Both seems to accomplish same thing but first one doesn't use a hook.

When would I use router vs useRouter and what's the difference?

If one of them is planned to be deprecated, which one? Which should I use?

**Comment by dani-mp** [comment]
Yeah, it seems weird that both APIs exist. Docs don't say anything about it.

**Comment by AntonioA1998** [comment]
Seems this still is the case, any answers?

**Reply by CoreyBovalina** [reply]
From my experience in using router vs useRouter in an app with a nested stack in tab layout, router will just push the tab stack over top itself and another stack. Whereas using useRouter properly pushes back to the tab stack instead of creating another.
